### PR TITLE
Convert index.js and ./components to ES6

### PR DESCRIPTION
**Pull request by AnNOtis**
see #3

### CODE REVIEW

File: components/NavBarContainer.js [hunk 1]
```diff
@@ -1,51 +1,55 @@
...
+    this.goBack = this.goBack.bind(this);
+    this.goForward = this.goForward.bind(this);
+    this.customAction = this.customAction.bind(this);
 
-  getInitialState: function() {
-    return {
+    this.state = {
       backButtonOpacity: 0,
-      previousRoute: {} // Keep previousRoute for smooth transitions
+      previousRoute: {}, // Keep previousRoute for smooth transitions
```
Comment (davidLeonardi) [inline components/NavBarContainer.js:21]: Trailing comma, please remove

File: components/NavBarContainer.js [hunk 2]
```diff
@@ -1,51 +1,55 @@
...
     };
-  },
+  }
 
-  componentWillReceiveProps: function(newProps) {
+  componentWillReceiveProps(newProps) {
     if (this.props && this.props.currentRoute.index !== newProps.currentRoute.index) {
       this.setState({
-        previousRoute: this.props.currentRoute
+        previousRoute: this.props.currentRoute,
```
Comment (davidLeonardi) [inline components/NavBarContainer.js:28]: Trailing comma, please remove

File: components/NavBarContainer.js [hunk 3]
```diff
@@ -107,23 +112,23 @@ var NavBarContainer = React.createClass({
...
 
-var styles = StyleSheet.create({
+const styles = StyleSheet.create({
   navbarContainer: {
     position: 'absolute',
     top: 0,
     left: 0,
     right: 0,
-    height: 64
+    height: 64,
```
Comment (davidLeonardi) [inline components/NavBarContainer.js:123]: Trailing comma, please remove

File: components/NavBarContainer.js [hunk 4]
```diff
@@ -107,23 +112,23 @@ var NavBarContainer = React.createClass({
...
   },
   navbarContainerHidden: {
     position: 'absolute',
     top: -64,
     left: 0,
     right: 0,
-    height: 64
-  }
+    height: 64,
```
Comment (davidLeonardi) [inline components/NavBarContainer.js:130]: Trailing comma, please remove

File: components/NavBarContainer.js [hunk 5]
```diff
@@ -107,23 +112,23 @@ var NavBarContainer = React.createClass({
...
+  },
```
Comment (davidLeonardi) [inline components/NavBarContainer.js:131]: Trailing comma, please remove

File: components/NavBarContent.js [hunk 6]
```diff
@@ -1,25 +1,29 @@
...
-
-var {
+const {
   StyleSheet,
   Text,
   View,
   Animated,
-  Easing
+  Easing,
+  Component,
```
Comment (davidLeonardi) [inline components/NavBarContent.js:10]: Trailing comma, please remove

File: components/NavBarContent.js [hunk 7]
```diff
@@ -30,43 +34,44 @@ var NavBarContent = React.createClass({
             fromValue: this.props.willDisappear ? 1 : 0,
             toValue: this.props.willDisappear ? 0 : 1,
             duration: 300,
-            easing: Easing.easeOutQuad
+            easing: Easing.easeOutQuad,
```
Comment (davidLeonardi) [inline components/NavBarContent.js:37]: Trailing comma, please remove

File: components/NavBarContent.js [hunk 8]
```diff
@@ -118,31 +138,41 @@ var NavBarContent = React.createClass({
...
     return (
-      <Animated.View style={[styles.navbar, transitionStyle, this.props.route.headerStyle,{borderBottomWidth: width, borderColor: color}, trans]}>
+      <Animated.View
+        style={
+          [
+            styles.navbar,
+            transitionStyle,
+            this.props.route.headerStyle,
+            { borderBottomWidth: width, borderColor: color },
+            trans,
```
Comment (davidLeonardi) [inline components/NavBarContent.js:163]: Trailing comma, please remove

File: components/NavBarContent.js [hunk 9]
```diff
@@ -152,7 +182,7 @@ var styles = StyleSheet.create({
     justifyContent: 'center',
     alignItems: 'center',
     flexDirection: 'row',
-    paddingTop: 13
+    paddingTop: 13,
```
Comment (davidLeonardi) [inline components/NavBarContent.js:185]: Trailing comma, please remove

File: components/NavBarContent.js [hunk 10]
```diff
@@ -169,18 +199,17 @@ var styles = StyleSheet.create({
   },
 
   alignLeft: {
-    alignItems: 'flex-start'
+    alignItems: 'flex-start',
```
Comment (davidLeonardi) [inline components/NavBarContent.js:202]: Trailing comma, please remove

File: components/NavBarContent.js [hunk 11]
```diff
@@ -169,18 +199,17 @@ var styles = StyleSheet.create({
...
   },
   alignRight: {
-    alignItems: 'flex-end'
+    alignItems: 'flex-end',
```
Comment (davidLeonardi) [inline components/NavBarContent.js:205]: Trailing comma, please remove

File: components/NavBarContent.js [hunk 12]
```diff
@@ -169,18 +199,17 @@ var styles = StyleSheet.create({
...
   },
   buttonTextLeft: {
     marginLeft: 10,
   },
   buttonTextRight: {
-    marginRight: 10
-  }
+    marginRight: 10,
```
Comment (davidLeonardi) [inline components/NavBarContent.js:211]: Trailing comma, please remove

File: components/NavBarContent.js [hunk 13]
```diff
@@ -169,18 +199,17 @@ var styles = StyleSheet.create({
...
+  },
```
Comment (davidLeonardi) [inline components/NavBarContent.js:212]: Trailing comma, please remove

File: components/NavButton.js [hunk 14]
```diff
@@ -1,30 +1,33 @@
...
 
-var React = require('react-native');
-
-var {
+const {
   StyleSheet,
   Text,
   View,
   TouchableHighlight,
+  Component,
```
Comment (davidLeonardi) [inline components/NavButton.js:8]: Trailing comma, please remove

File: components/NavButton.js [hunk 15]
```diff
@@ -44,8 +46,7 @@ var styles = StyleSheet.create({
     fontWeight: '600',
     textAlign: 'center',
     alignItems: 'center',
-  }
+  },
```
Comment (davidLeonardi) [inline components/NavButton.js:49]: Trailing comma, please remove

File: components/NavButton.js [hunk 16]
```diff
@@ -44,8 +46,7 @@ var styles = StyleSheet.create({
     fontWeight: '600',
     textAlign: 'center',
     alignItems: 'center',
```
Comment (sethx) [inline components/NavButton.js]: Trailing comma, please remove

Comment (davidLeonardi) [inline components/NavButton.js:48]: Trailing comma, please remove

File: index.js [hunk 17]
```diff
@@ -1,32 +1,38 @@
...
+    this.renderScene = this.renderScene.bind(this);
+    this.onDidFocus = this.onDidFocus.bind(this);
+    this.onWillFocus = this.onWillFocus.bind(this);
+
     this.state = {
       route: {
         name: null,
-        index: null
-      }
+        index: null,
```
Comment (davidLeonardi) [inline index.js:28]: Trailing comma, please remove

File: index.js [hunk 18]
```diff
@@ -1,32 +1,38 @@
...
+      },
```
Comment (davidLeonardi) [inline index.js:29]: Trailing comma, please remove

File: index.js [hunk 19]
```diff
@@ -161,44 +169,45 @@ class Router extends React.Component{
...
     );
   }
 }
 
-var styles = StyleSheet.create({
+const styles = StyleSheet.create({
   container: {
     flex: 1,
-    backgroundColor: '#FFFFFF'
+    backgroundColor: '#FFFFFF',
```
Comment (davidLeonardi) [inline index.js:209]: Trailing comma, please remove

File: index.js [hunk 20]
```diff
@@ -161,44 +169,45 @@ class Router extends React.Component{
...
   },
```
Comment (davidLeonardi) [inline index.js:210]: Trailing comma, please remove

**Comment by davidLeonardi** [comment]
Thanks for this contribution.
Sorry for being a codenazi about the trailing commas, but I think that trailing commas should not be allowed. Anyone else has an opinion on this? I get that it only used to break IE, but still i find it's not really correct JS.
@johannhof @AnNOtis @helielson  opinions wanted!

**Comment by AnNOtis** [comment]
Currently, our eslint configuration enforce trailing comma.
It's base on airbnb/javascript style guide.
More informations :point_right: https://github.com/airbnb/javascript#commas

**Comment by davidLeonardi** [comment]
My personal opinion on that is that it's wrong, and should be changed in the linting settings. This is probably one of the points where i disagree with the airbnb styleguide.
Citing from the interwebs:
"Using ES6 syntax implies full support for ES5. As well as that, today’s transpilers like babel remove the last comma anyway". 
So what this tells us, is that babel fixes this for us. Why not simply adhere to the language then?

**Comment by davidLeonardi** [comment]
Either way, i'd say we should match what ReactNative has.
Seems they allow trailing commas. :(
@AnNOtis do you have a clue what styleguide they adopt?

**Comment by davidLeonardi** [comment]
Anyhow, as trailing commas match RN's style, i'll consider this a go.

**Pull request merged**